tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/TokenQueryLogic.sol

```diff
@@ -57,57 +45,48 @@ abstract contract TokenQueryLogic is TokenHoldingAmount {
 			.identityToString(network, identity.value);
 
 		if (identityToStringSuccess) {
-			string memory url;
-			uint32[] memory networks = tokenNetworks[tokenName];
 			uint256 totalBalance = 0;
 
-			for (uint32 i = 0; i < networks.length; i++) {
-				// Check if this network has been queried
-				url = networkUrls[networks[i]];
-
-				if (!queriedNetworks[networks[i]]) {
-					string memory _tokenContractAddress = tokenAddresses[
-						tokenName
-					][networks[i]];
-					if (networks[i] == Web3Networks.BitcoinP2tr) {
-						uint256 balance = GeniidataClient.getTokenBalance(
-							secrets,
-							url,
-							identityString,
-							tokenName,
-							getTokenDecimals()
-						);
-						totalBalance += balance;
-					} else if (
-						networks[i] == Web3Networks.Bsc ||
-						networks[i] == Web3Networks.Ethereum
-					) {
-						(bool success, uint256 balance) = NoderealClient
-							.getTokenBalance(
-								url,
-								secrets,
-								_tokenContractAddress,
-								identityString
-							);
-						if (success) {
-							totalBalance += balance;
-						}
-					}
-					// Mark this network as queried
-					queriedNetworks[networks[i]] = true;
```

Comment:
Should `queriedNetworks` mapping be removed here? If the network is bsc and ethereum, what I mean is when execute param is :
```
identities = [

{

identity_type: 4,

value: "0xf7764F15Fd4F695fAD8dDd3e84F5293E6866FEFE",

networks: [7, 8],

},

]
```
in this case, will it be queried twice here?

Reply:
As you said, same identity could has different networks, but we need sum the token balance from different network.
e.g.
1) There has an evm identity A has 2 networks ethereum and bsc.
2) We need generate TokenHoldingAmount VC for token TRX.
3) The total balance of token TRX for identity A is: (TRX token balance in network ethereum) + (TRX token balance in network bsc).